src/sdam/monitor.rs

```diff
@@ -159,7 +158,7 @@ impl HeartbeatMonitor {
                         .map(|sd| sd.is_available())
                         .unwrap_or(false)
                 {
-                    self.handle_error(e.to_string(), topology, server).await;
+                    self.handle_error(e, topology, server).await;
```

Comment:
I needed to go back to regular `Error` in a few places since we need the ability to handle the error differently based on its characteristics, which would be hard/impossible with just the error message.

Reply:
Since we ended up deciding to keep Error cloneable, this seems fine.